contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/GobblinDistcpJob.java

```diff
@@ -0,0 +1,195 @@
+package voldemort.store.readonly.mr.azkaban;
+
+import java.util.Map;
+import java.util.List;
+import java.net.URI;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.permission.FsAction;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.apache.hadoop.mapred.JobConf;
+
+import gobblin.runtime.api.JobExecutionResult;
+import gobblin.runtime.embedded.EmbeddedGobblin;
+import gobblin.runtime.embedded.EmbeddedGobblinDistcp;
+
+import voldemort.utils.Props;
+import voldemort.utils.logging.PrefixedLogger;
+import voldemort.store.readonly.swapper.AdminStoreSwapper;
+
+import azkaban.jobExecutor.AbstractJob;
+
+
+public class GobblinDistcpJob extends AbstractJob {
+    static String source;
+    private final String destination;
+    private final Props props;
+
+    GobblinDistcpJob(String id, String source, String destination, Props props) {
+        super(id, PrefixedLogger.getLogger(AdminStoreSwapper.class.getName(), destination));
+        this.source = source;
+        this.destination = destination;
+        this.props = props;
+    }
+
+    public void run() throws Exception {
+        String cdnURL = pickCDN();
+
+        if (!cdnURL.isEmpty()) {
+            info("####################################");
+            info("              Distcp");
+            info("####################################");
+            info("Using CDN: " + cdnURL);
+            try {
+                String username = props.getString("env.USER", "unknownClient");
+                String pathPrefix = props.getString(VoldemortBuildAndPushJob.PUSH_CDN_PREFIX, "/tmp/VoldemortBnP/");
+                pathPrefix = pathPrefix.endsWith("/") ? pathPrefix : pathPrefix + "/";
+                String cdnDir = source.replaceAll(".*://.*?(?=/)", cdnURL + pathPrefix + username);
```

Comment:
Instead of using 'username' to compose the destination url, I will prefer to use storeName.
If we are using 'username', the CDN disk usage for one given store is not deterministic since multiple users could push data to the same store.